elasticdl/python/worker/allreduce_trainer.py

```diff
@@ -182,3 +184,96 @@ def export_saved_model(self, model_path):
 
     def get_model_version(self):
         return self._optimizer.iterations.numpy()
+
+
+class ElasticAllReduceController(object):
```

Comment:
Can we move the controllers into a separate python file?

Reply:
Yes, a good idea.